### PR TITLE
Feature/GitHub ci

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

File: .github/workflows/python_and_docker.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    runs-on: ubuntu-latest
+    container: docker:19.03.15
+
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Login to GitHub Packages
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
```
Comment (ThibsG) [inline .github/workflows/python_and_docker.yml:27]: reuse `${{ env.REGISTRY }}`?

File: .github/workflows/python_build.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3
+          cache: pip # caching pip dependencies
+
+      - name: Prerequisites
+        run: |
+          pip install maturin[patchelf]==0.14.9
+          sudo apt-get install --no-install-recommends -qq libclang-dev gcc-mingw-w64-x86-64
```
Comment (ThibsG) [inline .github/workflows/python_build.yml:97]: not a big deal, but maybe `sudo` is not required here

File: .github/workflows/python_and_docker.yml (hunk 1 above)
Comment (Manuthor) [reply .github/workflows/python_and_docker.yml:27]: thanks

File: .github/workflows/python_and_docker.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+name: Docker
+
+on:
+  workflow_call:
```
[the remark this answers is not in the document]
Comment (Manuthor) [reply .github/workflows/python_and_docker.yml:5]: it is required to call this workflow (as a reusable workflow). But you are right, calling a reusable workflow without parameters is not optimal

File: .github/workflows/python_build.yml (hunk 2 above)
Comment (Manuthor) [reply .github/workflows/python_build.yml:97]: I tried without sudo but it is unfortunately required :/

**Pull request closed without merging**